### PR TITLE
Add GetHashCode test for invariant mode

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
This is pending on the fix https://github.com/dotnet/coreclr/pull/16924

**Comment by tarekgh** [comment]
#28021

File: src/System.Globalization/tests/Invariant/InvariantMode.cs [hunk 1]
```diff
@@ -784,5 +785,14 @@ public void GetUnicode(string ascii, int index, int count, string expected)
             }
             Assert.Equal(expected, new IdnMapping().GetUnicode(ascii, index, count));
         }
+
+        [Fact]
+        public void TestHashing()
```
Comment (danmosemsft) [inline src/System.Globalization/tests/Invariant/InvariantMode.cs]: Better name? Maybe like `TestTurkishITurkishHashMatchesOrdinalHash`

Comment (tarekgh) [reply src/System.Globalization/tests/Invariant/InvariantMode.cs]: The intention here is not testing the Turkish but the intention is, in the invariant mode we'll not do cultural hashing and instead we'll do it ordinal.

**Comment by tarekgh** [comment]
@dotnet-bot test this please

**Comment by tarekgh** [comment]
test Windows x64 Debug Build please
test OSX x64 Debug Build please

**Comment by tarekgh** [comment]
The System.Net failure is not related and opened a tracking issue #28122

**Pull request merged**